pandas/core/indexing.py

```diff
@@ -186,33 +186,21 @@ def __setitem__(self, key, value):
         indexer = self._get_setitem_indexer(key)
         self._setitem_with_indexer(indexer, value)
 
-    def _has_valid_type(self, k, axis):
+    def _validate_key(self, k, axis):
         raise NotImplementedError()
```

Comment:
can you make this an AbstractClassError, and add a doc-string here

Reply:
> can you make this an AbstractClassError,

Pointer?

Reply:
AbstractMethoderror in pandas/core/base.py